### ISSUE TITLE
Packed repeating fields: error under MicroPython

### ISSUE THREAD

**Issue by peterhinch** [dogtopus/minipb#6]
This script runs under CPython 3.6.9 but fails under MicroPython:
```python
import minipb

data = {'txt' : [b'abc', b'def', b'ghi'],
       }
schema = (('txt', '#a'),
          )
w = minipb.Wire(schema)
tx = w.encode(data)
rx = w.decode(tx)
print(rx)
print('Length', len(tx))
```
This produces the following:
```
Traceback (most recent call last):
  File "<stdin>", line 7, in <module>
  File "minipb.py", line 465, in decode
  File "minipb.py", line 712, in decode_wire
  File "minipb.py", line 713, in <genexpr>
  File "minipb.py", line 541, in _break_down
AttributeError: 'BytesIO' object has no attribute 'tell'
>>> 
```
The same error occurs with `#U`. Replacing the # with + results in working code.

Possibly this can be documented rather than fixed. Using packed repeating fields results in minuscule savings: just a single byte under CPython in this test. I doubt there is a need for them in a MicroPython context.

**Comment by dogtopus** [comment]
>AttributeError: 'BytesIO' object has no attribute 'tell'

Hm that doesn't look good. I guess MicroPython misses more features than I anticipated.

Anyway one thing I would like to point out is https://github.com/dogtopus/minipb/wiki/Differences-between-MiniPB-and-Official-Protobuf#packed-stringsbytes

Maybe I should have a separate "best practice" documentation and link it from README just for some helpful redundancies.

**Comment by dogtopus** [comment]
Actually I just removed the unintentional field 15 limitation in 04c1c57 which uses tell() to determine whether or not the decoder is at the end of the stream. So the whole decode() is now broken under MicroPython because of that.

**Comment by dogtopus** [comment]
The general decoding issue should be fixed now with 63d5db5. However I would still not recommend using `#` with non-vint and non-fixed types simply because it might be gone if I eventually implement cross compatibility between packed and non-packed fields.
